apps/financroo-tpp/consent_commons.go

```diff
@@ -64,6 +70,82 @@ func (o *OBUKLoginURLBuilder) BuildLoginURL(consentID string, client acpclient.C
 		acpclient.WithPKCE())
 }
 
+type CDRLoginURLBuilder struct {
+	signingKey interface{}
+}
+
+func NewCDRLoginURLBuilder(config Config) (LoginURLBuilder, error) {
+	var bs []byte
+	var err error
+	var signingKey interface{}
+
+	if bs, err = os.ReadFile(config.KeyFile); err != nil {
+		return nil, errors.Wrapf(err, "failed to read request object signing key")
+	}
+
+	block, _ := pem.Decode(bs)
+
+	if signingKey, err = x509.ParsePKCS1PrivateKey(block.Bytes); err != nil {
+		return nil, errors.Wrapf(err, "failed to parse request object signing key")
+	}
+
+	return &CDRLoginURLBuilder{
+		signingKey,
+	}, nil
+}
+
+type ClaimRequests struct {
+	Userinfo map[string]*ClaimRequest `json:"userinfo"`
+	IDToken  map[string]*ClaimRequest `json:"id_token"`
+}
+
+type ClaimRequest struct {
+	Essential bool     `json:"essential"`
+	Value     string   `json:"value"`
+	Values    []string `json:"values"`
+}
+
+func (o *CDRLoginURLBuilder) BuildLoginURL(arrangementID string, client acpclient.Client) (authorizeURL string, csrf acpclient.CSRF, err error) {
+	var (
+		parsed      *url.URL
+		signedToken string
+	)
+
+	if authorizeURL, csrf, err = client.AuthorizeURL(acpclient.WithPKCE()); err != nil {
+		return authorizeURL, csrf, err
+	}
+
+	if parsed, err = url.Parse(authorizeURL); err != nil {
+		return authorizeURL, csrf, err
+	}
+
+	claims := jwt.MapClaims{
```

Comment:
Can't we use acp-client-go for this logic?

Reply:
I've made changes in acp-client-go so that we can use it for cdr: https://github.com/cloudentity/acp-client-go/pull/34